fpx/src/data/tests.rs

```diff
@@ -0,0 +1,172 @@
+use crate::api;
+use crate::api::models::SpanKind;
+use crate::data::util::Json;
+use crate::data::{migrations, RowsExt, Store};
+use libsql::params;
+use serde::Deserialize;
+use std::collections::BTreeMap;
+use test_log::test;
+use time::format_description::well_known::Rfc3339;
+use time::OffsetDateTime;
+use tokio::join;
+use tracing::info;
+
+/// Initialize a in memory database, and run the migrations on it.
+async fn create_test_database() -> Store {
+    let store = Store::in_memory().await.unwrap();
+
+    // Run migrations until the latest version
+    migrations::migrate(&store).await.unwrap();
+
+    store
+}
+
+#[tokio::test]
```

Comment:
maybe we can add compile tests like we had in autometrics to prevent misuse of write and read transactions?

Reply:
How would you suggest we do this? 

We could only create the "happy" path (ie read transaction for a read fn), since the invalid usage would prevent the compiler from finishing. And the happy path would be tested already by normal usage.

Reply:
yes we can just test the happy path as the compiler will then ensure it wont happen or we can go and use https://crates.io/crates/trybuild to test both happy path and compile error path. i especially like the later option as it'll ensure the stuff that should error actually does and that the error message actually matches our expectation. we have usage of it already in autometrics so we can use that as a template: https://github.com/autometrics-dev/autometrics-rs/blob/main/autometrics/tests/compilation.rs

Reply:
Okay that sounds like it could be useful. I will create a linear ticket for it and keep this PR as it is.